pallets/capacity/src/lib.rs

```diff
@@ -299,4 +346,41 @@ impl<T: Config> Pallet<T> {
 	fn calculate_capacity(amount: BalanceOf<T>) -> BalanceOf<T> {
 		amount
 	}
+
+	/// Decrease a staking account's active token and create an unlocking chunk to be thawed at some future block.
+	fn decrease_active_staking_balance(unstaker: &T::AccountId, amount: BalanceOf<T>) -> DispatchResult {
+		let mut staking_account_details = Self::get_staking_account_for(unstaker).ok_or(Error::<T>::NotStakingAccount)?;
+		ensure!(amount <= staking_account_details.active, Error::<T>::AmountToUnstakeExceedsAmountStaked);
+
+		let current_block: T::BlockNumber = frame_system::Pallet::<T>::block_number();
+		let thaw_at = current_block.saturating_add(T::BlockNumber::from(T::UnstakingThawPeriod::get()));
+
+		staking_account_details.decrease_by(amount, thaw_at)?;
+		Self::set_staking_account(&unstaker, &staking_account_details);
+
+		Ok(())
+	}
+
+	/// Reduce available capacity of target and return the amount of capacity reduction.
+	fn reduce_capacity(unstaker: &T::AccountId, target: MessageSourceId, amount: BalanceOf<T>) -> Result<BalanceOf<T>, DispatchError>  {
+		let mut staking_target_details = Self::get_target_for(&unstaker, &target).ok_or(Error::<T>::NotStakingAccount)?;
+		let mut capacity_details = Self::get_capacity_for(target).ok_or(Error::<T>::NotStakingAccount)?;
+		let capacity_reduction = Self::calculate_capacity_reduction(amount, capacity_details.total_tokens_staked, capacity_details.total_available)?;
+		staking_target_details.decrease_by(amount, capacity_reduction);
+		capacity_details.decrease_by(capacity_reduction, amount);
+
+		Self::set_capacity_for(target, capacity_details);
+		Self::set_target_details_for(unstaker, target, staking_target_details);
+
+		Ok(capacity_reduction)
+	}
+
+	/// Determine the capacity reduction when given total_capacity, unstaking_amount, and total_amount_staked.
+	fn calculate_capacity_reduction(unstaking_amount: BalanceOf<T>, total_amount_staked: BalanceOf<T>, total_capacity: BalanceOf<T>) -> Result<BalanceOf<T>, DispatchError> {
```

Comment:
I think it would be a good idea to add some tests to this one.

Reply:
Good catch, thanks @shannonwells 
We removed the checked_div function in favor of determining the rate as perbill